transitions/registration.js

```diff
@@ -333,14 +340,43 @@ module.exports = {
         var doc = options.doc,
             db = db || options.db;
 
-        transitionUtils.addUniqueId(db, doc, callback);
+        if (options.params.patient_id) {
+            var located = doc.fields[options.params.patient_id];
+
+            if (!located) {
+                transitionUtils.addRejectionMessage(
+                    doc,
+                    options.registrationConfig,
+                    'no_provided_patient_id');
+                return callback(null, true);
+            }
+
+            transitionUtils.isIdUnique(db, located, function(isUnique) {
+                if (isUnique) {
+                    doc.patient_id = located;
+                    callback();
+                } else {
+                    transitionUtils.addRejectionMessage(
+                        doc,
+                        options.registrationConfig,
+                        'provided_patient_id_not_unique');
+                    return callback(null, true);
```

Comment:
<nitpick>Inconsistent use of return with line 357 - either use return or `else` block for both. It's fine, but when I read it it makes me question whether I really understand the flow, so I have to read it twice.</nitpick>